pkg/tfbridge/property_path.go

```diff
@@ -58,6 +63,119 @@ func lookupSchemas(
 	return LookupSchemas(schemaPath, tfs, ps)
 }
 
+func combinePropertyMapKeys(
+	object1, object2 resource.PropertyMap,
+) map[resource.PropertyKey]struct{} {
+	combined := make(map[resource.PropertyKey]struct{})
+	for k := range object1 {
+		combined[k] = struct{}{}
+	}
+	for k := range object2 {
+		combined[k] = struct{}{}
+	}
+	return combined
+}
+
+// SkipChildrenError is an error that can be returned by the visitor to skip the children of the current step.
+type SkipChildrenError struct{}
+
+func (SkipChildrenError) Error() string {
+	return "skip children"
+}
+
+type twoPropertyValueVisitor func(path propertyPath, val1, val2 resource.PropertyValue) error
+
+// walkTwoPropertyValues walks the two property values and calls the visitor for each step.
+// It returns an error if the visitor returns an error other than SkipChildrenError.
+//
+// The visitor can return SkipChildrenError to skip the children of the current step.
+// In case the two values have different types, we walk both values, starting with val1.
+func walkTwoPropertyValues(
```

Comment:
This really is incomplete in ignoring the structure of PropertyValue grammar like Secret and Output and ResourceReference values that can have child PropertyValue. This may be actually OK for the intended application because we can strip secret and output wrappers prior to running this algo and still get acceptable results, but this is suspect for the purposes of generalizing out like TransformPropertyValue.

Reply:
Worth a doc comment? Or just handle it and recur into those? For a caller that cannot afford stripping secrets and outputs prior to doing a walk it is really unusable to walk into those containers.

Reply:
I actually find it a bit confusing that the other walker algorithm recurses into Secrets and Outputs. IMO these are not "children" but more of a wrapper, since both the child and the Secret/Output live at the same path.

I've added a comment to clarify this - it doesn't matter for this use case as Secrets and Outputs are stripped before the detailed diff happens.